### PR TITLE
Fixed tensor copy to remove UserWarning

### LINKED ISSUE #475: User Warning thrown when calling AveragePrecision
## 🐛 Bug

Using the AveragePrecision metric throws the following warning every time it is called:

UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).

### To Reproduce

1. Import AveragePrecision and create an object of this class
2. Call the object by passing in pred and target tensors

#### Code sample

```
import torch
from torchmetrics import AveragePrecision

a = torch.softmax(torch.randn(4, 4), dim=1)
b = torch.randint(0, 2, (4, 4))

ap = AveragePrecision(pos_label=1)
ap(a, b)
```

### Expected behavior
This warning should not be thrown. The recommended way of copying a tensor should be followed.

### Environment

- PyTorch Version: 1.8.2
- OS (e.g., Linux): Ubuntu 18
- How you installed PyTorch (`conda`, `pip`, source): conda
- Build command you used (if compiling from source):
- Python version: 3.8.5
- CUDA/cuDNN version: 11.1/8.0.5
- GPU models and configuration: Quadro RTX 8000
- Any other relevant information:

### Additional context

The problem is arising due to line 156 in torchmetrics/functional/classification/precision_recall_curve.py
The line currently reads:
```
thresholds = tensor(reversed(thresholds[sl]))
```
This problem should be fixed by changing that line to:
```
thresholds = reversed(thresholds[sl]).detach().clone()
```

### PR DESCRIPTION
**Pull request by nishprabhu**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
It changes a line of code in torchmetrics/functional/classification/precision_recall_curve.py to remove a UserWarning that was being thrown.

Fixes #475

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: torchmetrics/functional/classification/precision_recall_curve.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def _precision_recall_curve_compute_single_class(
 
     recall = torch.cat([reversed(recall[sl]), torch.zeros(1, dtype=recall.dtype, device=recall.device)])
 
-    thresholds = tensor(reversed(thresholds[sl]))
+    thresholds = reversed(thresholds[sl]).detach().clone()
```
Comment (SkafteNicki) [inline torchmetrics/functional/classification/precision_recall_curve.py:156]:
```suggestion
    thresholds = reversed(thresholds[sl]).detach().clone()  # type: ignore
```
